### DISCUSSION TITLE
Can I use the MDX files generated as components and import those in src/pages?

### DISCUSSION THREAD

**Discussion by santiagoejm** [PaloAltoNetworks/docusaurus-openapi-docs#376] [Q&A] [answered]
Hello Community! Hopefully this happens to somebody else before :)

What I am trying to do is treat the MDX files generated after running this plugin (`yarn docusaurus gen-api-docs all`) like any other component and import those in the React components that are in `src/pages`, but when I try it, I get this error:

<img width="685" alt="Screenshot" src="https://user-images.githubusercontent.com/68547995/209826720-33d12e20-5f1e-479f-888a-aeeb8c027ea6.png">

Please would somebody point me at the right direction of what am I missing here?
Cheers

**Comment by sserrata** [comment, marked as answer]
Hi @santiagoejm, the generated MDX would need to be wrapped in either `DocItem` or `ApiItem` to include the necessary providers. That said, if you wrap using `ApiItem` it will expect to receive props for the `ApiDemoPanel` portion, which are normally passed via the `api` frontmatter key.

**Reply by santiagoejm** [reply]
Hi @sserrata Thank you so much for taking the time to respond! There seems to be a quite straight forward solution, just that I do not see how to apply it. In my `docusaurus.config.js` file I have allow the use of the @theme/ApiItem, which from my understanding I would need to import in the React component where I currently am importing the .mdx file and wrap it, right? 

In my case does not work, I don't want to abuse of your time but the time is pressing on my side and I would really like to use this plugin, would you mind to take a look at this file where I want to import the MDX generated file [testing-branch](https://github.com/santiagoejm/eclipse-tractusx.github.io/blob/so-far/src/pages/BusinessKit/SoftwareDevelopmentView/index.js). as you can see in line 9 I am importing the .mdx and for now simply I just want to call it in line 106(currently is commented out, in the future I will place it in a toggle component and etc...), can you specify the declaration to import the Wrapper? if that is the case.

Thanks!

**Reply by sserrata** [reply]
Hi @santiagoejm, I assumed the provider error was the result of a missing Docusaurus provider. When I attempted to reproduce it locally I noticed that it was actually referring to the Redux `Provider` which is required by the `MimeTabs` component. That said, one workaround is to replace the `MimeTabs` component with the `Tabs` component.

1. Change `import MimeTabs from "@theme/MimeTabs";` to `import Tabs from "@theme/Tabs";`
2. Find and replace "MimeTabs" with "Tabs" (there should be only be 2 occurrences, opening and closing tags)

Let me know if this works. Thanks!

**Reply by sserrata** [reply]
The only caveat is that `ApiDemoPanel` won't/cannot be included, as it both requires the redux provider AND `frontmatter.api` props to render. It also won't render unless the page is wrapped in `ApiItem`, which, AFAIK, is probably not possible with a page.

**Reply by santiagoejm** [reply]
Hey @sserrata, thanks for the clarification! following your steps it actually work and I am able to place the .mdx component where I like to. Cheers to that!

It is indeed a shame that the `ApiDemoPanel` is not included, I'm gonna have to work around this for a bit, if by any chance comes to mind a possible solution, it will be very much appreciate :)

Thanks a lot, it was a good progress on my end!

**Reply by santiagoejm** [reply]
Hi @sserrata, sorry that I haven't resolved this discussion, but I am happy that I didn't because I have just one more question after rethinking how to implement this plugin.

This is a different related issue than the previous one, but I am just wondering if by default this plugin generate in the `sidebar` next to each item the METHOD icons GET, POST, PUT, DEL... like the example:

![Screenshot 2023-01-18 at 11 39 04](https://user-images.githubusercontent.com/68547995/213150775-f2894a38-b18e-45a0-a165-85efeaadcb93.png)

In my case locally is generating the mdx files and the `sidebar` items properly but without those nice icons GET, PUT, DEL... 

I am wondering if there is a missing info that I should add to the `.yaml` files from where the mdx is generated for each end point? I notice that the `sidebar` titles for each mdx is coming from the `operationId` property in .yaml Or it has to do with the `sidebarOptions/customProps` object, in which case I can not find the valid properties.

Thanks in advanced

**Reply by sserrata** [reply]
Please see this doc: https://docusaurus-openapi.tryingpan.dev/customization/styling

**Reply by santiagoejm** [reply]
@sserrata legend!

**Comment by tzvik15** [comment]
Good morning,

Is it possible you have the wrong Daniel for this?

Cheers,
